app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -147,24 +149,31 @@ class DebugInfoActivity: AppCompatActivity() {
     }
 
     /**
-     * Builds intent to view the problematic local event, task or contact at given Uri.
+     * Builds intent to view the problematic local resource at given Uri.
      *
-     * Note that only OpenTasks is supported as tasks provider. TasksOrg and jtxBoard
-     * do not support viewing tasks via intent-filter (yet). See also [at.bitfire.davdroid.sync.SyncNotificationManager.getLocalResourceUri]
+     * Note that the TasksOrg app does not support viewing tasks via intent-filter.
+     * @see [at.bitfire.davdroid.resource.LocalResource.getViewUri]
      */
     private fun buildViewLocalResourceIntent(uri: Uri): Intent? {
-        val activeTasksAuthority = tasksAppManager.get().currentProvider()?.authority
-        return when (uri.authority) {
-            ContactsContract.AUTHORITY ->
-                Intent(Intent.ACTION_VIEW).apply {
-                    setDataAndType(uri, ContactsContract.Contacts.CONTENT_ITEM_TYPE)
-                }
+        // Support ACTION_VIEW
+        val supportActionView = listOf(
+            CalendarContract.AUTHORITY, // any calendar app
+            JtxContract.JtxICalObject.VIEW_INTENT_HOST // jtx Board for journals, notes, tasks
+        )
+        if (uri.authority in supportActionView)
+            return Intent(Intent.ACTION_VIEW, uri)
 
-            CalendarContract.AUTHORITY, activeTasksAuthority ->
-                Intent(Intent.ACTION_VIEW, uri)
+        // Need ACTION_EDIT (OpenTasks crashes on using ACTION_VIEW)
+        if (uri.authority == TaskProvider.ProviderName.OpenTasks.authority) // OpenTasks app
+            return Intent(Intent.ACTION_EDIT, uri)
 
-            else -> null
-        }
+        // Need CONTENT_ITEM_TYPE to be set
+        if (uri.authority == ContactsContract.AUTHORITY) // any contacts app
+            return Intent(Intent.ACTION_VIEW).apply {
+                setDataAndType(uri, ContactsContract.Contacts.CONTENT_ITEM_TYPE)
+            }
```

Comment:
Minor style suggestion: since there are multiple branches that depend on `uri.authority`, maybe that would be easier to read?

~~~kotlin
return when (uri.authority) {
  in listOf("1", "2") -> ...
  "3" -> ...
  else -> null
}
~~~